--- slider-agent/src/main/python/agent/main.py
+++ slider-agent/src/main/python/agent/main.py
@@ -47,6 +47,18 @@ logFileName = "slider-agent.log"
 
 SERVER_STATUS_URL="https://{0}:{1}{2}"
 
+# Monkey patch SSL as long as https:// hosts are used without a signed cert
+import ssl
+
+try:
+    _create_unverified_https_context = ssl._create_unverified_context
+except AttributeError:
+    # Legacy Python that doesn't verify HTTPS certificates by default
+    pass
+else:
+    # Handle target environment that doesn't support HTTPS verification
+    ssl._create_default_https_context = _create_unverified_https_context
+
 
 def signal_handler(signum, frame):
   #we want the handler to run only for the agent process and not
