--- security.py.orig	2016-05-05 18:24:22.000000000 +0000
+++ security.py	2016-05-12 18:33:24.635788157 +0000
@@ -21,6 +21,7 @@
 import urlparse
 import socket
 import ssl
+import sys
 import os
 import logging
 import subprocess
@@ -36,6 +37,8 @@
 GEN_AGENT_KEY = 'openssl req -new -newkey rsa:1024 -nodes -keyout "%(keysdir)s'+os.sep+'%(hostname)s.key" '\
   '-subj /OU=%(hostname)s/ -out "%(keysdir)s'+os.sep+'%(hostname)s.csr"'

+if sys.version_info >= (2, 7, 9):
+  ssl._create_default_https_context = ssl._create_unverified_context

 class VerifiedHTTPSConnection(httplib.HTTPSConnection):
   """ Connecting using ssl wrapped sockets """